.github/workflows/c_release.yml

```diff
@@ -159,15 +156,13 @@ jobs:
         with:
           ref: c_release-${{github.run_number}}
       - name: Create tarball
+        working-directory: ./packages/c
         run: |
-          mkdir tarball
-          cd ./packages
-          mv c csshnpd-${{ github.ref_name }}
-          tar -cvzf ../tarball/csshnpd-${{ github.ref_name }}.tar.gz csshnpd-${{ github.ref_name }}
+          cmake --workflow --preset package-source
       - uses: actions/upload-artifact@4cec3d8aa04e39d1a68397de0c4cd6fb9dce8ec1 # v4.6.1
         with:
           name: csshnpd-src-${{github.ref_name}}-${{github.run_number}}-${{github.run_attempt}}
-          path: ./tarball/csshnpd-${{ github.ref_name }}.tar.gz
+          path: ./packages/c/build/release-static/noports-source.tar.gz
```

Comment:
I'd like to preserve the naming convention in terms of the tarball filename, and also the contents within it.

So csshnpd-c1.0.4.tar.gz rather than noports-source.tar.gz

And csshnpd-c1.0.4 rather than noports-source

Whether we do that with CPack cleverness, or just some script here doesn't bother me.

Reply:
Maybe I wasn't clear that both the tarball name and the root directory it contains need to follow the same convention